[bug] MSBuildDeps incorrectly tries to import .props for non-transitive header-only packages that are not generated

### Environment details

* Operating System+version: Windows 11
* Compiler+version: MSVC 17.8.6
* Conan version: 2.0.17
* Python version: 3.12.1


### Steps to reproduce

Create a conanfile for a MSBuild project as such:
```py
from conan import ConanFile
from conan.tools.microsoft import MSBuildDeps

class Pkg(ConanFile):
    settings = "os", "compiler", "arch", "build_type"
    
    def requirements(self):
        self.requires("scnlib/2.0.0")
     
    def generate(self):
        ms = MSBuildDeps(self)
        ms.generate()
```
Call conan install on the project with any msvc profile and output dir of your choice.
Look at the .props files written out to the output dir by conan MSBuildDeps: for example, for x64 release, in the file `conan_scnlib__scnlib_release_x64.props` there is a line `<Import Condition="'$(conan_fast_float_props_imported)' != 'True'" Project="conan_fast_float.props"/>`, but conan_fast_float.props does not exist.
This causes msbuild to fail when trying to build the project that imports the main conandeps.props generated by MSBuildDeps with `1>C:\testProj\conanprops\conan_scnlib__scnlib_release_x64.props(4,5): error MSB4019: The imported project "C:\testProj\conanprops\conan_fast_float.props" was not found. Confirm that the expression in the Import declaration "conan_fast_float.props" is correct, and that the file exists on disk.`
This is the same issue as here https://github.com/conan-io/conan/issues/15334 (and probably related to https://github.com/conan-io/conan/issues/14014) but is much easier/quicker to see with scnlib rather than opencv (which does exhibit the same issue in my case as well).


### Logs

conan install:
```
conan install . --output-folder=conanprops -pr:h=msvc193-x64-release -pr:b=msvc193-x64-release -vtrace

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=static
compiler.runtime_type=Release
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=static
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
Graph root
    conanfile.py: C:\testProj\conanfile.py
Requirements
    fast_float/6.1.0#85dc5bf9d3d6e605abf536818397baf3 - Cache
    scnlib/2.0.0#1001c947d4b617948955f7bd91dfe816 - Cache
    simdutf/4.0.5#484ca7bbfbbb2bd3db6289011f5ba613 - Cache
Build requirements
    cmake/3.28.1#92f79424d7b65b12a84a2180866c3a78 - Cache
Resolved version ranges
    cmake/[>=3.16 <4]: cmake/3.28.1

======== Computing necessary packages ========
Requirements
    fast_float/6.1.0#85dc5bf9d3d6e605abf536818397baf3:da39a3ee5e6b4b0d3255bfef95601890afd80709#69f6c6968a0f2f98bdffcadd07a3ad2d - Skip
    scnlib/2.0.0#1001c947d4b617948955f7bd91dfe816:260caab82f7f9a9b15225e54c089dc0403ae11eb#9732b7f31ad0ba2745ed3e082b277882 - Cache
    simdutf/4.0.5#484ca7bbfbbb2bd3db6289011f5ba613:a2e7df6f7bd606b2763e544f3a2c87dd6c478874#55f6770eed0b83432ab48f06caef5454 - Cache
Build requirements
    cmake/3.28.1#92f79424d7b65b12a84a2180866c3a78:522dcea5982a3f8a5b624c16477e47195da2f84f#29e6903b5006c2b2f81003c9ebf7551b - Skip

======== Installing packages ========
simdutf/4.0.5: Already installed! (1 of 2)
scnlib/2.0.0: Already installed! (2 of 2)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: scnlib/2.0.0

======== Finalizing install (deploy, generators) ========
conanfile.py: Calling generate()
conanfile.py: Generators folder: C:\testProj\conanprops
conanfile.py: Generating aggregated env files
conanfile.py: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
Install finished successfully
```
generated files:
```
ls .\conanprops\

        Directory: C:\testProj\conanprops


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a---        2024-02-08     04:39           2215   conan_scnlib__scnlib_release_x64.props
-a---        2024-02-08     04:39           1126   conan_scnlib__scnlib_vars_release_x64.props
-a---        2024-02-08     04:39            425   conan_scnlib__scnlib.props
-a---        2024-02-08     04:39            389   conan_scnlib.props
-a---        2024-02-08     04:39           1925   conan_simdutf_release_x64.props
-a---        2024-02-08     04:39            924   conan_simdutf_vars_release_x64.props
-a---        2024-02-08     04:39            404   conan_simdutf.props
-a---        2024-02-08     04:39             56   conanbuild.bat
-a---        2024-02-08     04:39            593   conanbuildenv-release-x86_64.bat
-a---        2024-02-08     04:39            272   conandeps.props
-a---        2024-02-08     04:39             54   conanrun.bat
-a---        2024-02-08     04:39            585   conanrunenv-release-x86_64.bat
-a---        2024-02-08     04:39             67   deactivate_conanbuild.bat
-a---        2024-02-08     04:39             65   deactivate_conanrun.bat
```

--- a/conan/tools/microsoft/msbuilddeps.py
+++ b/conan/tools/microsoft/msbuilddeps.py
@@ -315,6 +315,7 @@ class MSBuildDeps(object):
         condition = self._condition()
         dep_name = self._dep_name(dep, build)
         result = {}
+        pkg_deps = get_transitive_requires(self._conanfile, dep)  # only non-skipped dependencies
         if dep.cpp_info.has_components:
             pkg_aggregated_content = None
             for comp_name, comp_info in dep.cpp_info.components.items():
@@ -327,8 +328,9 @@ class MSBuildDeps(object):
                 public_deps = []  # To store the xml dependencies/file names
                 for required_pkg, required_comp in comp_info.parsed_requires():
                     if required_pkg is not None:  # Points to a component of a different package
-                        public_deps.append(required_pkg if required_pkg == required_comp
-                                           else "{}_{}".format(required_pkg, required_comp))
+                        if required_pkg in pkg_deps:  # The transitive dep might have been skipped
+                            public_deps.append(required_pkg if required_pkg == required_comp
+                                               else "{}_{}".format(required_pkg, required_comp))
                     else:  # Points to a component of same package
                         public_deps.append("{}_{}".format(dep_name, required_comp))
                 public_deps = [self._get_valid_xml_format(d) for d in public_deps]
@@ -348,7 +350,6 @@ class MSBuildDeps(object):
             vars_filename = "conan_%s_vars%s.props" % (dep_name, conf_name)
             activate_filename = "conan_%s%s.props" % (dep_name, conf_name)
             pkg_filename = "conan_%s.props" % dep_name
-            pkg_deps = get_transitive_requires(self._conanfile, dep)
             public_deps = [self._dep_name(d, build) for d in pkg_deps.values()]
 
             result[vars_filename] = self._vars_props_file(require, dep, dep_name, cpp_info,

--- a/conans/test/integration/toolchains/microsoft/test_msbuilddeps.py
+++ b/conans/test/integration/toolchains/microsoft/test_msbuilddeps.py
@@ -4,6 +4,7 @@ from xml.dom import minidom
 
 import pytest
 
+from conans.test.assets.genconanfile import GenConanfile
 from conans.test.utils.tools import TestClient
 
 
@@ -65,3 +66,57 @@ def test_msbuilddeps_format_names():
             minidom.parseString(content)
             counter += 1
     assert counter == 8
+
+
+class TestMSBuildDepsSkips:
+    # https://github.com/conan-io/conan/issues/15624
+    def test_msbuilddeps_skipped_deps(self):
+        c = TestClient()
+        c.save({"liba/conanfile.py": GenConanfile("liba", "0.1").with_package_type("header-library"),
+                "libb/conanfile.py": GenConanfile("libb", "0.1").with_package_type("static-library")
+                                                                .with_requires("liba/0.1"),
+                "app/conanfile.py": GenConanfile().with_requires("libb/0.1")
+                                                  .with_settings("arch", "build_type")})
+        c.run("create liba")
+        c.run("create libb")
+        c.run("install app -g MSBuildDeps")
+        assert not os.path.exists(os.path.join(c.current_folder, "app", "conan_liba.props"))
+        assert os.path.exists(os.path.join(c.current_folder, "app", "conan_libb.props"))
+        libb = c.load("app/conan_libb.props")
+        assert "conan_liba" not in libb
+        libb = c.load("app/conan_libb_release_x64.props")
+        assert "conan_liba" not in libb
+        libb = c.load("app/conan_libb_vars_release_x64.props")
+        assert "conan_liba" not in libb
+
+    def test_msbuilddeps_skipped_deps_components(self):
+        c = TestClient()
+        libb = textwrap.dedent("""
+            from conan import ConanFile
+            class Libb(ConanFile):
+                name = "libb"
+                version = "0.1"
+                package_type = "static-library"
+                requires = "liba/0.1"
+                def package_info(self):
+                    self.cpp_info.components["mycomp"].libs = ["mycomplib"]
+                    self.cpp_info.components["mycomp"].requires = ["liba::liba"]
+            """)
+
+        c.save({"liba/conanfile.py": GenConanfile("liba", "0.1").with_package_type("header-library"),
+                "libb/conanfile.py": libb,
+                "app/conanfile.py": GenConanfile().with_requires("libb/0.1")
+                                                  .with_settings("arch", "build_type")})
+        c.run("create liba")
+        c.run("create libb")
+        c.run("install app -g MSBuildDeps")
+        assert not os.path.exists(os.path.join(c.current_folder, "app", "conan_liba.props"))
+        assert os.path.exists(os.path.join(c.current_folder, "app", "conan_libb.props"))
+        libb = c.load("app/conan_libb.props")
+        assert "conan_liba" not in libb
+        libb = c.load("app/conan_libb_mycomp.props")
+        assert "conan_liba" not in libb
+        libb = c.load("app/conan_libb_mycomp_release_x64.props")
+        assert "conan_liba" not in libb
+        libb = c.load("app/conan_libb_mycomp_vars_release_x64.props")
+        assert "conan_liba" not in libb

